webapp/app/js/filters.js

```diff
@@ -39,11 +39,21 @@ treeherder.filter('stripHtml', function() {
 
 treeherder.filter('linkifyBugs', function() {
     return function(input) {
-        var re = new RegExp('(?:Bug (\\d+))', 'ig');
-        var str = input || '';
-        return str.replace(re,
-            '<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=$1" target="_blank" title="Bug $1">$1</a>'
-        );
+        if (input) {
+            var re = /.*-- (bug-)?([0-9]+)/i; //regex for matching bug number
+            var bug_matches = re.exec(input);
+            var bug_url = "https://bugzilla.mozilla.org/show_bug.cgi?id="+bug_matches[2];
+            var re = /PR#([0-9]+)/i; //regex for matching PR number
+            var pr_matches = re.exec(input);
+            var pr_url = "https://github.com/mozilla-b2g/gaia/pull/" + pr_matches[1];
+            var str = input || '';
+            var pr_hyperlink ='<a href="'+pr_url+'">'+pr_matches[1]+'</a>';
+            str = str.replace(pr_matches[1],pr_hyperlink);
+            var bug_hyperlink ='<a href="'+bug_url+'">'+bug_matches[2]+'</a>';
+            str = str.replace(bug_matches[2],bug_hyperlink);
```

Comment:
If the regex does not matches for either of the bug_matches and pr_matches, should it return an empty string?